packages/@aws-cdk/cfnspec/lib/schema/resource-type.ts

```diff
@@ -31,10 +34,42 @@ export interface ResourceType extends Documented {
 export interface TaggableResource extends ResourceType {
   Properties: {
     Tags: TagProperty;
+    UserPoolTags: TagProperty;
     [name: string]: Property;
   }
 }
 
+export interface TaggableResourceStandard extends TaggableResource {
+  Properties: {
+    Tags: TagPropertyStandard;
+    UserPoolTags: TagPropertyStandard;
+    [name: string]: Property;
+  }
+}
+
+export interface TaggableResourceAutoScalingGroup extends TaggableResource {
+  Properties: {
+    Tags: TagPropertyAutoScalingGroup;
+    UserPoolTags: TagPropertyAutoScalingGroup;
```

Comment:
Does it make sense that an autoscaling group will have a `UserPoolTags` property?

Reply:
Why do we need all these interfaces? Not sure I understand this change

Reply:
you are totally right -- now that I have the `keyOf` working, I don't need these.